### DISCUSSION TITLE
Is it possible to run fltk with winit

### DISCUSSION THREAD

**Discussion by GodGotzi** [fltk-rs/fltk-rs#1508] [Q&A]
If its possible please give me some hints how I could do that

**Comment by MoAlyousef** [comment]
Short answer: Unfortunately no it's not possible.

Long answer: Technically it's possible. It would require changes in FLTK itself. Adding a winit driver and instructing FLTK to use that for windowing and to draw inside. The reverse is pointless since winit itself doesn't do drawing, so asking winit to take a raw fltk window handle would just be so that winit can handle the event loop. Anyways, both approaches would require changes in either FLTK or winit.

--------------

Some crates use the raw-window-handle crate which exposes HasRawWindowHandle trait, and this allows them to take a window (whether winit's or fltk's or other) and do their own drawing inside them, for example the [pixels crate](https://github.com/parasyte/pixels):
- https://github.com/parasyte/pixels/tree/main/examples/minimal-fltk
- https://github.com/parasyte/pixels/tree/main/examples/minimal-winit

**Reply by GodGotzi** [reply]
Thanks for the quick response, I will have a look at that
